src/python/pants/goal/stats_aggregator.py

```diff
@@ -55,13 +56,30 @@ class StatsAggregatorSubsystem(Subsystem):
         ),
         advanced=True,
     )
+    output_file = StrOption(
+        default=None,
+        metavar="<path>",
+        help="Output the stats to this file. If unspecified, outputs to stdout.",
+    )
+
+
+def log_or_write_to_file(output_file: Optional[str], text: str, logger: logging.Logger) -> None:
+    """Send text to the stdout or write to the output file."""
+    if output_file:
+        with open(output_file, "w") as fh:
+            fh.write(text)
```

Comment:
I think it could be a nice thing to log the fact that the stats data was written to file x here.

Reply:
Done!